src/Modules/Trinity.DynamicCluster/Trinity.DynamicCluster/Storage/DynamicRemoteStorage.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Trinity.Diagnostics;
+using Trinity.DynamicCluster.Consensus;
+using Trinity.Network;
+using Trinity.Network.Messaging;
+using Trinity.Storage;
+
+namespace Trinity.DynamicCluster
+{
+    class DynamicRemoteStorage : RemoteStorage
+    {
+
+        public DynamicRemoteStorage(ServerInfo server_info, int connPerServer, MemoryCloud mc, int pid, bool nonblocking) 
```

Comment:
Leo's solution is to publish the instance GUID together with ServerInfo, so we do not have to set the id later with a query. Not sure if it's a good idea to add this requirement to an adapter implementation...